clusterman/util.py

```diff
@@ -74,13 +74,16 @@ class ClustermanResources(NamedTuple):
 
     @staticmethod
     def from_instance_type(instance_type: str) -> "ClustermanResources":
-        resources = EC2_INSTANCE_TYPES[instance_type]
-        return ClustermanResources(
-            cpus=resources.cpus,
-            mem=resources.mem * 1024,  # AWS metadata for RAM is in GB
-            disk=(resources.disk or DEFAULT_VOLUME_SIZE_GB) * 1024,  # AWS metadata for disk is in GB
-            gpus=resources.gpus,
-        )
+        resources = get_instance_type(instance_type)
+        if resources is not None:
+            return ClustermanResources(
+                cpus=resources.cpus,
+                mem=resources.mem * 1024,  # AWS metadata for RAM is in GB
+                disk=(resources.disk or DEFAULT_VOLUME_SIZE_GB) * 1024,  # AWS metadata for disk is in GB
+                gpus=resources.gpus,
+            )
+        else:
+            raise Exception(f"instance type {instance_type} is invalid.")
```

Comment:
maybe we should move non-exis/invalid exception to get_instance_type function. And it would help us to keep other behaviours same as before. 

- get_instance_type replaced EC2_INSTANCE_TYPES map. we got exception if we request non-exist type. 
- get_market_resources returned InstanceResources, but now it's optional.